### PR TITLE
Add generic member_data to simplify operator governance

### PR DESCRIPTION
**Pull request by eddyashton**
As described in #1597.

This adds a `member_data` field for each member, in the same pattern as `user_data`, storing arbitrary JSON for the governance scripts to interpret how they wish. `operator_gov.lua` is updated to use this, creating an object with `is_operator = true` for operating members, rather than hardcoding them in the constitution. 

#1597 mentioned exposing the proposer's ID to the constitution, but I don't think this is actually necessary. We don't currently block proposals on any grounds, so we'd have to add that if we wanted to gate the _creation_ of specific proposals to members/non-members. Instead we care about the operatoricity of each _voter_. In the case of proposals submitted by operators (which are freely submitted like any other members'), their single yes vote is sufficient to pass an 'operator proposal' (one of the constitution-defined calls), but this works purely by vote-iteration and doesn't need to consider the proposer separately.

### CODE REVIEW

File: tests/memberclient.py [hunk 1]
```diff
@@ -38,8 +38,12 @@ def test_set_recovery_threshold(network, args, recovery_threshold=None):
 def test_add_member(network, args):
     primary, _ = network.find_primary()
 
+    member_data = {"name": "test_added_member", "roles": ["OWNER", "CONTRIBUTOR"]}
```
Comment (jumaffre) [inline tests/memberclient.py:41]: Super minor: When reading this, I genuinely thought that `OWNER` and `CONTRIBUTOR` were actual roles with specific semantics in CCF. For anyone (e.g. CCF operators) reading this test in isolation (e.g. to find examples of governance operations), could we replace these with dummy strings so that it's clear that these roles don't actually exist?

File: src/node/rpc/test/member_voting_test.cpp [hunk 2]
```diff
@@ -1478,9 +1491,60 @@ DOCTEST_TEST_CASE("Passing operator vote")
     DOCTEST_CHECK(proposer_vote != votes.end());
     DOCTEST_CHECK(proposer_vote->second == vote_for);
   }
+
+  auto new_operator_kp = tls::make_key_pair();
```
Comment (jumaffre) [inline src/node/rpc/test/member_voting_test.cpp:1495]: Should we also add tests for the negative paths, i.e. operator member trying to add or retire a non-operator member?

**Review by jumaffre — APPROVED** [review]

**Review by achamayou — APPROVED** [review]

File: src/node/rpc/test/member_voting_test.cpp (hunk 2 above)
Comment (eddyashton) [reply src/node/rpc/test/member_voting_test.cpp:1495]: Good call - I've added a few negative tests as well.
